core/tests/protractor_desktop/learnerDashboard.js

```diff
@@ -153,19 +208,234 @@ describe('Learner dashboard functionality', function() {
+    await topicEditorPage.addSubtopic(
+      'Subtopic TASV1', 'subtopic-tasv-one', Constants.TEST_SVG_PATH,
+      'Subtopic content');
+    await topicEditorPage.saveTopic('Added subtopic.');
+    await topicEditorPage.navigateToTopicEditorTab();
+    await topicEditorPage.navigateToReassignModal();
+    await topicEditorPage.dragSkillToSubtopic('Skill TASV1', 0);
+    await topicEditorPage.saveRearrangedSkills();
+    await topicEditorPage.saveTopic('Added skill to subtopic.');
+    await topicEditorPage.updateMetaTagContent('topic meta tag');
+    await topicEditorPage.updatePageTitleFragment('topic page title');
+    await topicEditorPage.togglePracticeTab();
+    await topicEditorPage.saveTopic('Added thumbnail.');
+    await topicEditorPage.publishTopic();
+    await topicsAndSkillsDashboardPage.editTopic('Topic TASV1');
+    await topicEditorPage.createStory(
+      'Story TASV1', 'story-player-tasv-one', 'Story description',
+      Constants.TEST_SVG_PATH);
+    await storyEditorPage.updateMetaTagContent('story meta tag');
+    for (var i = 0; i < 3; i++) {
+      await storyEditorPage.createNewChapter(
+        `Chapter ${i}`, dummyExplorationIds[i], Constants.TEST_SVG_PATH);
+      await storyEditorPage.navigateToChapterWithName(`Chapter ${i}`);
+      await storyEditorPage.changeNodeDescription('Chapter description');
+      await storyEditorPage.changeNodeOutline(
+        await forms.toRichText(`outline ${i}`));
+      await storyEditorPage.navigateToStoryEditorTab();
+    }
+    await storyEditorPage.saveStory('First save');
+    await storyEditorPage.publishStory();
+    await topicAndStoryViewerPage.get(
+      'math', 'topic-tasv-one', 'story-player-tasv-one');
+    await topicAndStoryViewerPage.expectCompletedLessonCountToBe(0);
+    await topicAndStoryViewerPage.expectUncompletedLessonCountToBe(3);
+    await topicAndStoryViewerPage.goToChapterIndex(0);
+    await explorationPlayerPage.submitAnswer('Continue', null);
+    await topicAndStoryViewerPage.get(
+      'math', 'topic-tasv-one', 'story-player-tasv-one');
+    await topicAndStoryViewerPage.expectCompletedLessonCountToBe(1);
+    await topicAndStoryViewerPage.expectUncompletedLessonCountToBe(2);
+    await topicAndStoryViewerPage.goToChapterIndex(1);
+    await explorationPlayerPage.submitAnswer('Continue', null);
+    await topicAndStoryViewerPage.get(
+      'math', 'topic-tasv-one', 'story-player-tasv-one');
+    await topicAndStoryViewerPage.expectCompletedLessonCountToBe(2);
+    await topicAndStoryViewerPage.expectUncompletedLessonCountToBe(1);
+    await topicAndStoryViewerPage.goToChapterIndex(2);
+    await explorationPlayerPage.submitAnswer('Continue', null);
+    await topicAndStoryViewerPage.get(
+      'math', 'topic-tasv-one', 'story-player-tasv-one');
+    await topicAndStoryViewerPage.expectCompletedLessonCountToBe(3);
+    await topicAndStoryViewerPage.expectUncompletedLessonCountToBe(0);
+    await learnerDashboardPage.get();
```

Comment:
Ditto, write status of server above here. Would be easier later on, if we wanted to expand the e2e tests.

Reply:
Done.